discovery-infra/test_infra/helper_classes/nodes.py

```diff
@@ -0,0 +1,121 @@
+import logging
+import json
+from typing import Dict
+import random
+from munch import Munch
+from test_infra import consts
+from test_infra.controllers.node_controllers.node import Node
+from test_infra.tools.concurrently import run_in_parallel
+
+
+class NodeMapping(object):
+
+    def __init__(self, node, cluster_host):
+        self.name = node.name
+        self.node = node
+        self.cluster_host = cluster_host
+
+
+class Nodes(object):
+
+    def __init__(self, node_controller, private_ssh_key_path):
+        self.controller = node_controller
+        self.private_ssh_key_path = private_ssh_key_path
+        self._nodes = None
+        self._nodes_as_dict = None
+
+    @property
+    def nodes(self):
+        if not self._nodes:
+            self._nodes = self.list()
+        return self._nodes
+
+    def __len__(self):
+        return len(self.nodes)
+
+    def get_masters(self):
+        return [node for node in self.nodes if node.is_master_in_name()]
+
+    @property
+    def nodes_as_dict(self):
+        if not self._nodes_as_dict:
+            self._nodes_as_dict = {node.name: node for node in self.nodes}
+        return self._nodes_as_dict
+
+    def list(self):
```

Comment:
I don't see why we need both list() and nodes. Maybe this can be a private method?